src/util.ts

```diff
@@ -144,4 +145,31 @@ export function parseAzureResourceId(resourceId: string): { [key: string]: strin
     }
 
     return result;
-}
\ No newline at end of file
+}
+
+export async function showWorkspaceFoldersQuickPick(placeHolderString: string): Promise<vscode.WorkspaceFolder> {
+    const folderQuickPickItems = vscode.workspace.workspaceFolders.map((value) => {
```

Comment:
Don't you still have to check if 'vscode.workspace.workspaceFolders' is defined here? That's what I meant when I said move the error handling logic into this function

Reply:
If there are no opened workspaceFolders, VS Code shows the user "There are no entries to pick from" as displayed above.  This will result in the user having to cancel, so we shouldn't have to worry about error handling.  Our error was actually never shown to the user.